mobile/library/common/internal_engine.cc

```diff
@@ -19,6 +21,17 @@ constexpr absl::Duration ENGINE_RUNNING_TIMEOUT = absl::Seconds(30);
 // Google DNS address used for IPv6 probes.
 constexpr absl::string_view IPV6_PROBE_ADDRESS = "2001:4860:4860::8888";
 constexpr uint32_t IPV6_PROBE_PORT = 53;
+
+// There is only one shared static Logger::Context instance for all Envoy Mobile engines.
+// This helps avoid issues on Logger::Context destruction when the previous saved context
+// could be activated in a thread-unsafe manner.
+void initOnceLoggerContext(const OptionsImplBase& options) {
+  static Thread::MutexBasicLockable* log_lock = new Thread::MutexBasicLockable();
```

Comment:
nit: MUTABLE_CONSTRUCT_ON_FIRST_USE if we can make that work, just for consistency.